server/container_create.go

```diff
@@ -163,6 +164,27 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
 	return processArgs, nil
 }
 
+// addOCIHooks look for hooks programs installed in hooksDirPath and add them to spec
+func addOCIHooks(specgen *generate.Generator, hooksDirPath string , configFile string) error {
+	// find any hooks executables
+	if _, err := os.Stat(hooksDirPath); os.IsNotExist(err) {
+		return nil
+	}
+	hooks, err := ioutil.ReadDir(hooksDirPath)
+	if err != nil {
+		return err
+	}
+	for _, item := range hooks {
+		if item.Mode().IsRegular() {
+			hook := filepath.Join(hooksDirPath, item.Name())
+			specgen.AddPreStartHook(hook, []string{hook, "prestart", configFile})
```

Comment:
We could get the config file from the bundlePath that is sent as part of state over stdin to the hooks.